test/runtests.jl

```diff
@@ -12,6 +12,14 @@ else
     throwing_function(bt) = bt[1]
 end
 
+const rseed = Ref(Base.GLOBAL_RNG)  # to get new random directories (see #24445)
```

Comment:
Any reason not to use a `RandomDevice` object here? it doesn't need seeding, and is used internally in a call to `srand(::MersenneTwister)`. Actually, `randtmp() = joinpath(tempdir(), randstring(RandomDevice(), 10))` should work (unless called too often and too quickly, as `RandomDevice()` opens a file ("/dev/urandom") which is not freed immediately).

Reply:
Didn't know about RandomDevice, that seems useful.